### PR TITLE
record span and breadcrumb when Django opens db connection

### LINKED ISSUE #1249: Instrument new database connections in Django
Sentry [instruments calls to `cursor.execute` and `cursor.executemany`](https://github.com/getsentry/sentry-python/blob/ed4ba68cad42ebfbab162b37bf7edad25ebeae55/sentry_sdk/integrations/django/__init__.py#L476-L515), but misses other database calls when Django opens a connection.

When Django opens a connection, it can make a few queries to configure timezones, auto commit settings and isolation settings. 

I think we could instrument `DatabaseWrapper.connect` and maybe add more instrumentation to log the initial queries Django makes to initialize a connection.

https://github.com/django/django/blob/8d9827c06ce1592cca111e7eafb9ebe0153104ef/django/db/backends/base/base.py#L197-L200

### PR DESCRIPTION
**Pull request by chdsbd**
Here's a screenshot of a trace where Django opens a new connection

Screen Shot 2021-11-14 at 9 04 39 PM

fixes #1249

### CODE REVIEW

File: sentry_sdk/integrations/django/__init__.py [hunk 1]
```diff
@@ -510,6 +513,19 @@ def executemany(self, sql, param_list):
...
+        # type: (BaseDatabaseWrapper) -> None
+        hub = Hub.current
+        if hub.get_integration(DjangoIntegration) is None:
+            return real_connect(self)
+
+        with capture_internal_exceptions():
+            hub.add_breadcrumb(message="connect", category="query")
+
+        with hub.start_span(op="db", description="connect"):
+            return real_connect(self)
```
Comment (chdsbd) [inline sentry_sdk/integrations/django/__init__.py:526]: not sure what the message or description should be

File: tests/integrations/django/test_basic.py [hunk 2]
```diff
@@ -19,19 +21,24 @@
...
 from sentry_sdk import capture_message, capture_exception, configure_scope
 from sentry_sdk.integrations.django import DjangoIntegration
+from functools import partial
 
 from tests.integrations.django.myapp.wsgi import application
 
 # Hack to prevent from experimental feature introduced in version `4.3.0` in `pytest-django` that
 # requires explicit database allow from failing the test
-pytest_mark_django_db_decorator = pytest.mark.django_db
+pytest_mark_django_db_decorator = partial(pytest.mark.django_db)
```
Comment (chdsbd) [inline tests/integrations/django/test_basic.py:30]: I updated this so I could pass transaction=False

**Comment by AbhiPrasad** [comment]
Will wait for tests match against the django matrix to see if it works with all versions, but LGTM from an initial glance. Thank you for your contribution!

**Comment by AbhiPrasad** [comment]
The test failures are due to `sanic`, which we fixed. Mind rebasing on master?

**Comment by chdsbd** [comment]
@AbhiPrasad I updated the branch

**Review by AbhiPrasad — COMMENTED** [review]
Seem like there some test failures, mind taking a look?

```
platform linux -- Python 3.8.12, pytest-6.2.5, py-1.11.0, pluggy-0.13.1
cachedir: .tox/py3.8-django-3.2/.pytest_cache
django: settings: tests.integrations.django.myapp.settings (from ini)
rootdir: /home/runner/work/sentry-python/sentry-python, configfile: pytest.ini
plugins: asyncio-0.16.0, hypothesis-6.24.5, localserver-0.5.0, forked-1.1.3, django-4.4.0, cov-2.8.1
collected 51 items
........F.
tests/integrations/django/test_basic.py ............Xx................   [ 54%]
tests/integrations/django/test_transactions.py .....                     [ 64%]
tests/integrations/django/asgi/test_asgi.py .......

=================================== FAILURES ===================================
__________________________ test_django_connect_trace ___________________________
tests/integrations/django/test_basic.py:428: in test_django_connect_trace
    assert res.status_code == 200
E   AttributeError: 'tuple' object has no attribute 'status_code'
```

**Comment by chdsbd** [comment]
@AbhiPrasad I think I finally fixed it. I was incorrectly using the test client like the Django Rest client.

**Comment by AbhiPrasad** [comment]
Awesome, thanks for sticking with this!!

**Pull request merged**